pkg/log/context.go

```diff
@@ -36,3 +58,40 @@ func goid() string {
 	idField := strings.Fields(strings.TrimPrefix(string(buf[:n]), "goroutine "))[0]
 	return idField
 }
+
+func paramKey(name string) string {
+	return PARAM + name
+}
+
+func GetParamFromContext(ctx context.Context, key string) (string, string) {
+	val := ctx.Value(contextKey(paramKey(key)))
+	if val == nil {
+		return key, ""
+	} else {
+		return key, val.(string)
+	}
+}
+
+//openapi middleware handler that takes URL params from the path and put them
+//on the context for further use (mainly by logs). parameters are prefixed to avoid
+//conflicts (for example: cluster_id --> PARAM$cluster_id)
+func ContextHandler() func(http.Handler) http.Handler {
+
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			//extract the Route from the request
+			mr := middleware.MatchedRouteFrom(r)
+			//set the parameters on the request's context
+			ctx := r.Context()
+			if mr != nil {
+				for _, p := range mr.Params {
```

Comment:
Why copy all the params to context?
We don't really need that.
Instead of iterating we can just check if the values we want exist and only copy them

Reply:
I thought that since route params are important  (usually meaningful indices) its good idea to copy them all (there aren't that many) so if we want to add another index to the logs we would have it handy. and it saves a lot of if-then clauses

Reply:
I don't understand what if-then clauses?
Why not just directly get the params we are interested in?